custom_components/tuya_local/devices/casdon_td_pro_2.yaml

```diff
@@ -0,0 +1,206 @@
+name: Casdon TD Pro 2
+products:
+  - id: wab6oustdukujpkn
+    name: 蒸烤箱TD-pro二代
+primary_entity:
+  entity: switch
+  icon: "mdi:power"
+  name: Switch
+  dps:
+    - id: 1
+      type: boolean
+      name: switch
+
+secondary_entities:
+  - entity: switch
+    name: Start
+    icon: "mdi:play-pause"
+    dps:
+      - id: 2
+        name: switch
+        type: boolean
+  - entity: number
+    name: cook_temperature
+    dps:
+      - id: 7
+        name: value
+        type: integer
+        unit: C
+        range:
+          min: 0
+          max: 500
+  - entity: sensor
+    name: temp_current
+    class: temperature
+    dps:
+      - id: 8
+        name: sensor
+        type: integer
+        unit: C
+        readonly: true
+        range:
+          min: 0
+          max: 500
+  - entity: number
+    name: appointment_time
+    dps:
+      - id: 9
+        name: value
+        type: integer
+        optional: true
+        range:
+          min: 0
+          max: 1440
+        unit: min
+        class: measurement
+  - entity: number
+    icon: "mdi:timer-edit"
+    name: cook_time
+    dps:
+      - id: 10
+        name: value
+        type: integer
+        range:
+          min: 0
+          max: 1440
+        unit: min
+        class: measurement
+  - entity: sensor
+    icon: "mdi:timer"
+    name: remain_time
+    dps:
+      - id: 11
+        name: sensor
+        type: integer
+        readonly: true
+        range:
+          min: 0
+          max: 1440
+        unit: min
+        class: measurement
+  - entity: sensor
+    name: fault_message
+    category: diagnostic
+    class: enum
+    dps:
+      - id: 13
+        type: bitfield
+        name: sensor
+        mapping:
+          - value: OK
+          - dps_val: 0
+            value: OK
+          - dps_val: 1
+            value: "er1：腔体高温保护(腔体传感器温度高于290°C)"
+          - dps_val: 2
+            value: "er2：蒸发盘高温保护(蒸发盘传感器温度高于200°C)"
+          - dps_val: 4
+            value: "er3：腔体低温保护(连续工作5分钟温度低于35°C)"
+          - dps_val: 8
+            value: "er4：蒸发盘低温保护(连续工作5分钟温度低于35°C)"
+          - dps_val: 16
+            value: "er5：传感器连接出故障(腔体或蒸发盘传感器开路)"
+          - dps_val: 32
+            value: "er6：传感器感应出故障(腔体或蒸发盘传感器短路)"
+          - dps_val: 64
+            value: "er7：显示板通信故障"
+          - dps_val: 128
+            value: "er8：水箱开关（CN7）未闭合或者蒸发盘NTC（CN3）温度高于125℃并持续时间达25s后，缺水提示，并停止工作。"
+  - entity: binary_sensor
+    name: fault
```

Comment:
Instead of name, use `class: problem` for HA to display appropriately.

Reply:
Okay